bed/compare.go

```diff
@@ -159,6 +159,24 @@ func CompareChromEndByChrom(a Bed, b Bed) int {
 	}
 	return CompareChromEnd(a, b)
 }
+//CompareDistance compares beds by chromStart and chromEnd and returns the minimum distance between the two beds.
+func CompareDistance(a Bed, b Bed) int {
+	chromComp := strings.Compare(a.Chrom, b.Chrom)
+	if chromComp != 0 {
```

Comment:
you could rearrange this so you can return the chromComp at the end:

if chromComp == 0 {
     if Overlap (a, b) {
         if a.ChromStart...{
}
        if a.ChromStart...{
}
}
else {
return chromComp (or whatever tells the user that the beds aren't on the same chrom
}
return var (you could store the int from the comparisons in this var instead of returning them above or something there are options)
}

Reply:
I like the idea with the chromComp. The var part isn't necessary, is it?

Reply:
just a way to have a return there if you don't have anything else im sure someone will have a way around it tho

Reply:
For what it's worth, I like the way you have it set up already. This is and example of the 'guard clause' we were talking about yesterday in code review. The early return saves you from doing more comparisons than needed, and helps reduce nested `if` statements which I always find hard to keep track of. The caveat is that you end up with multiple return's throughout the code which some people find jarring. Both are valid styles so no right answer here.